### ISSUE TITLE
CSS attribute selectors are removed even when whitelisted

### ISSUE THREAD

**Issue by Juneezee** [FullHuman/purgecss#277]
PurgeCSS version: 2.0.5

### style.css
```css
.button[disabled] {
  opacity: .5;
}
```

### postcss.config.js
```js
const purgecss = require('@fullhuman/postcss-purgecss')

module.exports = {
  plugins: [
    purgecss({
      ...
      whitelistPatterns: [/.*\[disabled].*/],
    })
  ]
}
```

## Expected Behavior
`.button[disabled]` is not removed 

## Actual Behavior
`.button[disabled]` is removed

**Comment by simllll** [comment]
I experienced the same issue,
actually we upgraded to 2.x .. and I was missing some styles, the extractors are valid, and do actually work, so I tried whitelisting the css class explicity.. but style is still gone. Any idea what the root cause could be of this behaviour?

**Comment by simllll** [comment]
Tried to debug it, 
I use scoped styles in vue, therefore my first issue was that "company-colors" as whitelist is not sufficient, as the css style gets rewritten to something like .company-colors[data-v-06391632].
Using a whitelist pattern, "solves" it.

But could it be that purgecss 1.x behavioud differently in the matter of css classes ending with a ` [ ]`? It seems purgecss 1.x keeps the class names regardless of the selector afterwards. purgecss 2.x doesn't. Maybe a side effect that was pretty helpful which got fixed in the meantime? Can we reenable this behaviour of keeping class names regardless of the selector. Or is there a way that I get the "full" css class names with the right selectors out of my build process? (Using purgecss webpack plugin)

**Comment by simllll** [comment]
One more update, it is defienitly related to https://github.com/FullHuman/purgecss/commit/d7f719d7c2683fdf4c6a3e91dd499e0afc1300c4#diff-ab0e95239dfd1a3bb06b9cd803c90c43

e.g. https://github.com/FullHuman/purgecss/commit/d7f719d7c2683fdf4c6a3e91dd499e0afc1300c4#diff-ab0e95239dfd1a3bb06b9cd803c90c43R539 this line confuses me a lot, and changing it back to return true does actually fix my issue (not the tests though).
But I can't really follow why this is a variable, and not just return true; if it is whistled, why would we check for anything else? Also if there is a match,  the for loop could reset the flag in the next run back to false?

**Comment by simllll** [comment]
Alright, I got closer to the actual issue.

I found out that the attributes are now more important than before ;). I had following regex:
![image](https://user-images.githubusercontent.com/5757263/73661645-aed00e00-469a-11ea-8e8d-6bafb396511b.png)
and assumed this was the correct behavior, but as the attributes are not handled separately, they were assumed that they do not exist.
Therefore chaging the regex to this:
![image](https://user-images.githubusercontent.com/5757263/73661784-f8205d80-469a-11ea-9c12-220a93e0d707.png)
solved a part of this deal (changing my regex from [A-z0-9-:\\/]+ to [A-Za-z0-9-:\\/]+.)

Unfortunately I couldn't find any way how to get the [data-v-*] attributes that nuxt / vue adds during build step, so my only idea to get things working again with scoped css is now to whitelist all data-vs:
`		whitelistPatterns: [/data-v-.*/],`

Well it turned out, here is the conenction back to the original issue of this ticket. Whitelisting of attributes isn't currently working. I've created a PR #279 to fix this.
After this PR has landed, following should work (beware to NOT add [ ])
`      whitelist: ['disabled'],`

regards and sorry for so many comments and text ;) thought maybe it helps to reproduce
